lib/ansible/module_utils/ansible_tower.py

```diff
@@ -83,6 +93,30 @@ def tower_check_mode(module):
             module.fail_json(changed=False, msg='Failed check mode: {0}'.format(excinfo))
 
 
+def sanitise_and_merge_variables(*args):
+    '''tower_auth_config attempts  to parse variables in json/yaml format including from a file.
+    If multiple arguments are provided they are merged left to right with the right having priority.
+    Returns a json encoded string
+    '''
+    result = {}
+    for var in args:
+        if isinstance(var, string_types):
+            if var.startswith('@'):
+                filename = os.path.expanduser(var[1:])
+                with open(filename, 'r') as f:
+                    var = f.read()
+            var = yaml.safe_load(var)
```

Comment:
I'm worried about correctly parsing the null case from server variables if `merge_variables` is true.

```python
>>> print yaml.safe_load('')
None
>>> print yaml.safe_load('---')
None
```

In this case, we wind up with None when an empty dict was expected. A cheap fix would be converting `elif`->`if` when you convert None into {}

Reply:
I actually stumbled upon this exact issue while using the merge_variables data. tower-cli was returning an empty string for the variables.

I changed the function to this to fix but haven't made the commit yet:
```
def sanitise_and_merge_variables(*args):
    '''tower_auth_config attempts  to parse variables in json/yaml format including from a file.
    If multiple arguments are provided they are merged left to right with the right having priority.
    Returns a json encoded string
    '''
    result = {}
    for var in args:
        if isinstance(var, string_types) and var != '':
            if var.startswith('@'):
                filename = os.path.expanduser(var[1:])
                with open(filename, 'r') as f:
                    var = f.read()
            var = yaml.safe_load(var)
        elif type(var) == dict:
            pass
        elif var is None or var == '':
            var = {}
        else:
            raise TypeError("Unexpected variable type")

        result = dict_merge(result, var)
    return json.dumps(result)
```